apps/nextjs/src/env/client.mjs

```diff
@@ -5,7 +5,7 @@ const _clientEnv = clientSchema.safeParse(clientEnv);
 
```

Comment:
This is unrelated to `apps/web`, but we can have consistency in env var checks by adding it to `apps/extension` as well.

The extension uses plasmo (parcel under the hood) and we cannot throw, but would have to replace the throw with a `process.exit(1)` to make it terminate the program during the build step.

Something like:

```diff
/**
 * 1. Specify your client-side environment variables schema here.
 * This way you can ensure the app isn't built with invalid env vars.
 * To expose them to the client, prefix them with `PLASMO_PUBLIC_`.
 */
export const clientSchema = z.object({
  PLASMO_PUBLIC_WEB_URL: z.string(),
});

// 2. this is a bit annoying, but maybe needs to be done?
export const clientEnv = {
  PLASMO_PUBLIC_WEB_URL: process.env.PLASMO_PUBLIC_WEB_URL,
};

// 3. parse the clientEnv from above according to our schema from (1)
const _clientEnv = clientSchema.safeParse(clientEnv);

if (!_clientEnv.success) {
  console.error(
    "❌ Invalid environment variables:\n",
    ...formatErrors(_clientEnv.error.format()),// some util
  );
-   throw new Error("Invalid environment variables");
+  process.exit(1)
}

// 4. additionally check if there's any key that's not prefixed with PLASMO_
 for (let key of Object.keys(_clientEnv.data)) {
   if (!key.startsWith("PLASMO_PUBLIC_")) {
     console.warn("❌ Invalid public environment variable name:", key)

-      throw new Error("Invalid public environment variable name")
+     process.exit(1)
   }
 }

```

I haven't implemented it because I'm not very familiar with the `process` api, are you?

Would be nice to have similar logs in the console as we have with `apps/web` for `extension` (parcel) and `outlook` (vite), too.